### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by ievgenaverianov**

### CODE REVIEW

File: Timer_JS_5-6.html [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    <meta charset="UTF-8">
+    <title>Timer Homework #5-6</title>
+    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0/css/bootstrap.min.css" integrity="sha384-Gn5384xqQ1aoWXA+058RXPxPg6fy4IWvTNh0E263XmFcJlSAwiGgFAW/dAiS6JXm" crossorigin="anonymous">
+    <link rel="stylesheet" type="text/css" href="css/Timer_JS_5-6.css">
+</head>
+<body>
+    <div class="timer">
+    <h1 class="timestring">00:00:00.000</h1>
+    <div class="buttons">
+        <button class="startbtn btn btn-success">start</button>
```
Comment (proprofondo) [inline Timer_JS_5-6.html]: use camelCase for better readability

File: js/Timer_JS_5-6.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
+var oTimer = {
+
+  iStartTime: 0,
+  iTimeNow: 0,
+  iPrintTime: 0,
+  iStopTime: 0,
+  iSplitTime: 0,
+  timerId: 0,
```
Comment (proprofondo) [inline js/Timer_JS_5-6.js]: iTimerId

File: js/Timer_JS_5-6.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+  bActiveTimer: false,
+
+
+  /**
+   * sets timer's time of start
+   */
+  startTimer: function () {
+    iStartTime = Date.now() - this.iPrintTime;
```
Comment (proprofondo) [inline js/Timer_JS_5-6.js]: you are using global variable instead of object property here. use this.iStartTime

File: js/Timer_JS_5-6.js [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+    var iTimeNow = Date.now();
+    return iTimeNow - iStartTime;
+  },
+
+  /**
+   * prints time every 1 msec since timer was started
+   */
+  printTime: function () {
+    this.startTimer();
+    this.getTime();
```
Comment (proprofondo) [inline js/Timer_JS_5-6.js]: why to call this method?

File: js/Timer_JS_5-6.js [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+    timerId = setInterval(function () {
```
Comment (proprofondo) [inline js/Timer_JS_5-6.js]: global var

File: js/Timer_JS_5-6.js [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+      var mlsec = oTimer.getTime();
```
Comment (proprofondo) [inline js/Timer_JS_5-6.js]: iMlsec

File: js/Timer_JS_5-6.js [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+      var sec = new Date(mlsec);
```
Comment (proprofondo) [inline js/Timer_JS_5-6.js]: make the name self-descriptive

File: js/Timer_JS_5-6.js [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+  /**
+   * pauses timer and prints to the new string below the time since timer's start till pause
+   */
+  pauseTimer: function () {
+    clearInterval(timerId);
+    this.iPrintTime = this.getTime();
+    iPausedTime = document.createElement('li');
+    var mlsec = oTimer.getTime() - this.iStopTime;
+    var sec = new Date(mlsec);
+    iPausedTime.innerHTML = 'Stop: ' + oTimer.checkTime((sec.getHours() - 2)) + ':' + oTimer.checkTime(sec.getMinutes()) + ':' + oTimer.checkTime(sec.getSeconds()) + '.' + oTimer.checkTimeMsec(sec.getMilliseconds());
```
Comment (proprofondo) [inline js/Timer_JS_5-6.js]: formatting can be done like separate method to avoid code duplication

File: js/Timer_JS_5-6.js [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+    document.querySelector('.startbtn').style.display = 'none';
+    document.querySelector('.pausebtn').style.display = 'inline-block';
+  },
+
+  /**
+   * pauses timer and prints to the new string below the time since timer's start till pause
+   */
+  pauseTimer: function () {
+    clearInterval(timerId);
+    this.iPrintTime = this.getTime();
```
Comment (proprofondo) [inline js/Timer_JS_5-6.js]: All of the 4 calls of `getTime()` can return you different result. If you need same moment in all cases call the method once and store the result to local var

File: js/Timer_JS_5-6.js [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+    document.querySelector('.startbtn').style.display = 'inline-block';
+    document.querySelector('.pausebtn').style.display = 'none';
+  },
+
+  /**
+   * splits time while timer is running
+   */
+  splitTime: function () {
+    if (this.bActiveTimer) {
+      var iSplitedTime = document.createElement('li');
```
Comment (proprofondo) [inline js/Timer_JS_5-6.js]: DOM element is object. would be oSplitTimeEl

File: js/Timer_JS_5-6.js [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+      sTimeList.removeChild(sTimeList.firstChild);
+    }
+  },
+
+  /**
+   * sets hours, minutes and seconds with leading zero
+   * @param i - counter
+   * @returns {*} - hours, minutes and seconds
+   */
+  checkTime: function (i){
```
Comment (proprofondo) [inline js/Timer_JS_5-6.js]: 1. this method not only checks but also transforms the value. please rename
2. you mix returned type that can cause unpredictable behavior in a calling method. hint: always return `sPrefix + i` to get string result.

File: js/Timer_JS_5-6.js [hunk 12]
```diff
@@ -0,0 +1,136 @@
...
+    }
+    return i;
+  },
+
+  /**
+   * sets miliseconds with leading zero
+   * @param i - counter
+   * @returns {*} - miliseconds
+   */
+  checkTimeMsec: function(i) {
```
Comment (proprofondo) [inline js/Timer_JS_5-6.js]: this method can be combined with previous one to smth like `addLeadingZeros(iValue, iNumberOfDigits)`. the most exact verb for this logic is probably ensureLeadingZeros

**Review by proprofondo — CHANGES_REQUESTED** [review]

File: Timer_JS_5-6.html (hunk 1 above)
Comment (ievgenaverianov) [reply Timer_JS_5-6.html]: done

File: js/Timer_JS_5-6.js (hunk 2 above)
Comment (ievgenaverianov) [reply js/Timer_JS_5-6.js]: done

File: js/Timer_JS_5-6.js (hunk 3 above)
Comment (ievgenaverianov) [reply js/Timer_JS_5-6.js]: done

File: js/Timer_JS_5-6.js (hunk 4 above)
Comment (ievgenaverianov) [reply js/Timer_JS_5-6.js]: done - line deleted

File: js/Timer_JS_5-6.js (hunk 5 above)
Comment (ievgenaverianov) [reply js/Timer_JS_5-6.js]: replaced with this.iTimerId

File: js/Timer_JS_5-6.js (hunk 9 above)
Comment (ievgenaverianov) [reply js/Timer_JS_5-6.js]: fixed - other 3 calls were replaced with this.iPrintTime

File: js/Timer_JS_5-6.js (hunk 10 above)
Comment (ievgenaverianov) [reply js/Timer_JS_5-6.js]: replaced with oSplitedTimeLine

File: js/Timer_JS_5-6.js (hunk 8 above)
Comment (ievgenaverianov) [reply js/Timer_JS_5-6.js]: done

File: js/Timer_JS_5-6.js (hunk 11 above)
Comment (ievgenaverianov) [reply js/Timer_JS_5-6.js]: done

File: js/Timer_JS_5-6.js (hunk 12 above)
Comment (ievgenaverianov) [reply js/Timer_JS_5-6.js]: methods combined into ensureLeadingZeros method

**Review by proprofondo — APPROVED** [review]

**Pull request merged**